### ISSUE TITLE
PanAnimation setView doesn't always fire moveend

### ISSUE THREAD

**Issue by foundryspatial-duncan** [Leaflet/Leaflet#3655]
The version of `map.setView()` in map.PanAnimation.js has [some cases](https://github.com/Leaflet/Leaflet/blob/v0.7.3/src/map/anim/Map.PanAnimation.js#L29) where it uses the PanAnimation version of `map.panBy()` and finishes without firing a `moveend` event.

The `map.setView()` in the core [map.js](https://github.com/Leaflet/Leaflet/blob/v0.7.3/src/map/Map.js#L60) file always calls `_resetView()` and therefore always fires a `moveend`. Shouldn't it always fire `moveend`? 

I'm referring to 0.7.3 but it looks like it's the same in 1.0b1

**Comment by mourner** [comment]
Animated `panBy` should always end with a `moveend`. If it's not the case, please set up a minimal JSFiddle test case that reproduces the issue.

**Comment by foundryspatial-duncan** [comment]
Sure, in this example I've set up a basic map with one marker. The marker is slightly away from the center of the map, such that `setView` & `panTo` wouldn't have to animate. Try clicking both buttons repeatedly and see that only _resetView emits `moveend` every time. 
http://jsfiddle.net/Lq0v2t9a/
(open your console before running the fiddle)

**Comment by mourner** [comment]
Looking at the fiddle, I see that moveend doesn't fire if no movement happens, but fires if there's a view reset. I think that's fine, I can't think of a use case where someone would have an issue with this. So I'm inclined to close unless someone proves me wrong.

**Comment by foundryspatial-duncan** [comment]
It was giving me trouble where I had a large number of markers, and some code that would center the map on a given marker. Lacking a callback parameter for `panTo()`, I used `map.once('moveend',...)` to execute code once I was sure the map was re-centered on the marker. But in some rare cases where two markers are very close together, the map wouldn't animate and the `moveend` was never emitted, so my `.once()` block wouldn't get executed until the the map actually *did* move later on.
In the case above, I worked around it by changing `panTo()` to `setView()` with the `{reset: true}` option.

I hope that makes sense. Is there perhaps a better way than watching for `moveend` to guarantee callback-like behavior from `panTo()`?
